### PR TITLE
Port rostopic bw

### PR DESCRIPTION
**Pull request by yechun1**
Port rostopic bw from ROS1 to ROS2, enable the tool to display bandwidth used by topic.

Fixed the issue: https://github.com/ros2/ros2cli/issues/132
This patch requires ros2/rclpy#242

### CODE REVIEW

File: ros2topic/ros2topic/verb/bw.py [hunk 1]
```diff
@@ -1,9 +1,8 @@
-#!/usr/bin/env python
-# Software License Agreement (BSD License)
-#
 # Copyright (c) 2008, Willow Garage, Inc.
 # All rights reserved.
 #
+# Software License Agreement (BSD License 2.0)
```
[the remark this answers is not in the document]
Comment (dirk-thomas) [reply ros2topic/ros2topic/verb/bw.py]: Sorry, my brain parsed this as "Apache 2.0". Where is 2.0 coming from though? Anyway I would suggest to leave this untouched.

File: ros2topic/ros2topic/verb/bw.py [hunk 2]
```diff
@@ -34,2113 +33,133 @@
...
-
-def _resource_name_package(name):
-    """
-    pkg/typeName -> pkg, typeName -> None
-    
-    :param name: package resource name, e.g. 'std_msgs/String', ``str``
-    :returns: package name of resource, ``str``
-    """    
-    if not '/' in name:
-        return
```
[the remark this answers is not in the document]
Comment (LucidOne) [reply ros2topic/ros2topic/verb/bw.py]: Can you check the c:coffee:ntext here? It looks like this was removed by the diff.

[the remark this answers is not in the document]
Comment (LucidOne) [reply ros2topic/ros2topic/verb/bw.py]: same

[the remark this answers is not in the document]
Comment (yechun1) [reply ros2topic/ros2topic/verb/bw.py]: seems `def _resource_name_package()` is only used by `ros2topic pub`, removed here for `ros2topic bw`.

**Comment by yechun1** [comment]
> Would it be possible to add some tests just to verify the basic behavior of the `bw` verb?

@mjcarroll I try to add test code but meet a issue that `colcon test` will not stop. I think the `bw` is a monitor, this will lead the test run into loop. Do you have any idea based with below test code?
```
def test_bw_publishers_subscribers():
    args = Namespace()
    args.topic = '/test_ros2_topic_cli'
    args.window = 100
    s = StringIO()
    with redirect_stdout(s):
        bw_verb = BwVerb()
        bw_verb.main(args=args)
        expected_output = _generate_expected_output(0, 0, 0, 0, args.window)
        assert expected_output == s.getvalue().splitlines()
```

File: ros2topic/ros2topic/verb/bw.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (dirk-thomas) [reply ros2topic/ros2topic/verb/bw.py]: > Based on original license and change BSP License to BSP License 2.0.

Thanks for describing it. In order to satisfy the linter this sounds like a good choice.

> Apply the same license with info/pub/echo.

Can you point out which exact files you are referring to? All the three verbs seem to be new implementations under the Apache license without copying code from ROS 1.

[the remark this answers is not in the document]
Comment (yechun1) [reply ros2topic/ros2topic/verb/bw.py]: That points to python files under verb folder, so they are different. As "bw" is ported based on ros1 code, I'll keep `bw` with original license and use "BSP License 2.0" to satisfy the linter. thanks.

File: ros2topic/ros2topic/verb/bw.py [hunk 3]
```diff
@@ -34,2113 +33,133 @@
...
-    table_width = sum(col_widths) + 3 * (len(header) - 1)
-    n_rows = len(cols[header[0]])
-    body = '\n'.join('   '.join(cols[h][i] for h in header) for i in xrange(n_rows))
-    table = '{header}\n{hline}\n{body}\n'.format(
-        header='   '.join(header_aligned), hline='=' * table_width, body=body)
-    return table
+    def add_arguments(self, parser, cli_name):
+        arg = parser.add_argument(
+            'topic',
+            help='Topic name to be calcurated the delay')
```
[the remark this answers is not in the document]
Comment (yechun1) [reply ros2topic/ros2topic/verb/bw.py]: fixed by c94f22e

File: ros2topic/ros2topic/verb/bw.py [hunk 4]
```diff
@@ -34,2113 +33,133 @@
...
+            include_hidden_topics_key='include_hidden_topics')
+        parser.add_argument(
+            '--window', '-w', type=unsigned_int, default=DEFAULT_WINDOW_SIZE,
+            help='window size, in # of messages, for calculating rate '
+                 '(default: %d)' % DEFAULT_WINDOW_SIZE, metavar='WINDOW')
 
-def _sleep(duration):
-    rospy.rostime.wallsleep(duration)
+    def main(self, *, args):
+        return main(args)
```
[the remark this answers is not in the document]
Comment (yechun1) [reply ros2topic/ros2topic/verb/bw.py]: fixed in c94f22e

File: ros2topic/ros2topic/verb/bw.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (yechun1) [reply ros2topic/ros2topic/verb/bw.py]: thanks, fixed in c94f22e

**Comment by mjcarroll** [comment]
Hey @yechun1 , sorry for the delay.  Can you rebase and --signoff your commits to appease the DCO and I will run a CI round on this?

**Comment by yechun1** [comment]
Hi @mjcarroll , just rebased code and re-submitted. 
Sorry I'm not familiar with test code, and did not add tests to verify the basic behavior of the bw verb.

**Comment by mjcarroll** [comment]
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/6716/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/3030/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/5519/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/6546/)

**Comment by mjcarroll** [comment]
Thanks for the contribution!

**Comment by yechun1** [comment]
thanks for approval and merge.

**Pull request merged**